webui/react/src/components/TableActionBar.tsx

```diff
@@ -154,39 +145,52 @@ const TableActionBar: React.FC<Props> = ({
     useModal(ExperimentTensorBoardModal);
   const isMobile = useMobile();
   const { openToast } = useToast();
-  const experimentIds = useMemo(
-    () => Array.from(selectedExperimentIds.keys()),
-    [selectedExperimentIds],
-  );
+
+  const [experiments, setExperiments] = useState<Loadable<BulkExperimentItem>[]>([]);
+
+  const fetchExperimentsByIds = useCallback(async (selectedIds: number[]) => {
+    const CHUNK_SIZE = 80; // match largest pageSizeOption used for Pagination components
+    const chunkedExperimentIds = _.chunk(selectedIds, CHUNK_SIZE);
+    const responses = await Promise.all(
+      chunkedExperimentIds.map(async (ids) => {
+        const response = await getExperiments({
+          experimentIdFilter: {
+            incl: ids,
+          },
+        });
+        return response.experiments;
+      }),
+    );
+    const fetchedExperiments = responses.reduce((acc, experiments) => {
+      return acc.concat(experiments);
```

Comment:
nit: [Javascript Array.push is 945x faster than Array.concat](https://uilicious.com/blog/javascript-array-push-is-945x-faster-than-array-concat/#:~:text=To%20merge%20arrays%20of%20size,is%20945x%20faster%20than%20concat%20!)
det repo doesnt use `concat` much anyways, so `push`?

```suggestion
      acc.push(...experiments);
      return acc;
```